pkg/gateway/gateway.go

```diff
@@ -339,9 +339,19 @@ func (g *gatewayType) onStoppedLeading() {
 
 	logger.Info("Controllers stopped")
 
-	if err := g.gatewayPod.SetHALabels(context.Background(), subv1.HAStatusPassive); err != nil {
-		logger.Warningf("Error updating pod label to passive: %s", err)
-	}
+	msgLogged := atomic.Bool{}
+	ctx := context.Background()
+	_ = wait.PollUntilContextCancel(ctx, 100*time.Millisecond, true, func(ctx context.Context) (bool, error) {
+		if err := g.gatewayPod.SetHALabels(ctx, subv1.HAStatusPassive); err != nil {
+			if msgLogged.CompareAndSwap(false, true) {
+				logger.Warningf("Error updating pod label to passive: %s", err)
```

Comment:
This has the potential to end up being confusing: since the operation is retried, a failure would result in a warning in the logs, which might be resolved by a subsequent retry — but nothing in the logs would indicate that. It would be better to only log an error if the pod label can’t be updated at all.